components/ingest-service/projectupdater/manager.go

```diff
@@ -52,6 +51,26 @@ func NewManager(client backend.Client, authzProjectsClient iam_v2.ProjectsClient
 	}
 }
 
+func (manager *Manager) Cancel(projectUpdateID string) {
+	switch manager.state {
+	case notRunningState:
+		// do nothing job is not running
+	case runningState:
+		if manager.projectUpdateID == projectUpdateID {
+			logrus.Debugf("Cancelling project tag update for ID %q elasticsearch task ID %q",
+				manager.projectUpdateID, manager.esJobID)
+			manager.client.JobCancel(context.Background(), manager.esJobID)
```

Comment:
Tell elasticsearch to cancel the update process. Let the `waitingForJobToComplete` process publish a complete status when the job is stopped.